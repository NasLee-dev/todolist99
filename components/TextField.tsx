interface TextFieldProps extends React.InputHTMLAttributes<HTMLInputElement> {
  errorMsg?: string
  label: string
}

const TextField = ({ label, errorMsg, ...props }: TextFieldProps) => {
  return (
    <div className="flex flex-col w-full">
      <label className="text-sm font-semibold text-gray-600">{label}</label>
      <input
        {...props}
        className="w-full border border-gray-300 rounded-md p-3 focus:outline-none focus:border-blue-500 transition duration-200 mt-2"
        placeholder="Type here"
      />
      {errorMsg && <span className="text-red-500 text-sm">{errorMsg}</span>}
    </div>
  )
}

export default TextField
