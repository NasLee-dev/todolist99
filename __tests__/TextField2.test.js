import { render, screen } from '@testing-library/react'
import TextField from '../components/TextField'

describe('TextField Component', () => {
  it('renders correctly', () => {
    const { container } = render(<TextField label="Name" />)
    expect(container).toMatchSnapshot() // 스냅샷 테스트
  })

  it('displays the label', () => {
    render(<TextField label="Name" />)
    expect(screen.getByLabelText('Name')).toBeInTheDocument() // label이 잘 렌더링되었는지 확인
  })

  it('renders an input field', () => {
    render(<TextField label="Name" />)
    expect(screen.getByRole('textbox')).toBeInTheDocument() // textbox 역할의 input이 존재하는지 확인
  })

  it('shows the correct placeholder', () => {
    render(<TextField label="Name" placeholder="Type here" />)
    expect(screen.getByPlaceholderText('Type here')).toBeInTheDocument() // placeholder가 잘 설정되었는지 확인
  })

  it('sets the value prop correctly', () => {
    render(<TextField label="Name" value="John" />)
    expect(screen.getByRole('textbox')).toHaveValue('John') // input의 value가 잘 설정되었는지 확인
  })

  it('disables the input when disabled prop is true', () => {
    render(<TextField label="Name" disabled />)
    expect(screen.getByRole('textbox')).toBeDisabled() // input이 비활성 상태인지 확인
  })

  it('renders the error message', () => {
    render(<TextField label="Name" errorMsg="Required" />)
    expect(screen.getByText('Required')).toBeInTheDocument() // 에러 메시지가 잘 렌더링되는지 확인
  })

  it('does not render error message when there is none', () => {
    render(<TextField label="Name" />)
    expect(screen.queryByText('Required')).not.toBeInTheDocument() // 에러 메시지가 렌더링되지 않아야 함
  })
})
