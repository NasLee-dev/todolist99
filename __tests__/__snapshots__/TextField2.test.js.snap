// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component renders correctly 1`] = `
<div>
  <div
    class="flex flex-col w-full"
  >
    <label
      class="text-sm font-semibold text-gray-600"
    >
      Name
    </label>
    <input
      class="w-full border border-gray-300 rounded-md p-3 focus:outline-none focus:border-blue-500 transition duration-200 mt-2"
      placeholder="Type here"
    />
  </div>
</div>
`;
